### PR TITLE
Add new printer drivers

### PR DESCRIPTION
**Pull request by apmasell**
And rename existing ones to be more understandable

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java [hunk 1]
```diff
@@ -98,9 +67,70 @@ public boolean print(byte[] content, JsonNode configuration) {
...
+    public boolean print(byte[] content, JsonNode configuration) {
+      return sendFtpFile(configuration.get("host").asText(), "admin", configuration.get("password").asText(),
+          null, content.hashCode() + ".ZPL", content, configuration);
+    }
+
   };
   private static final Logger log = LoggerFactory.getLogger(Backend.class);
 
+  protected static boolean sendFtpFile(String host, String username, String password, String directory, String filename, byte[] content,
+      JsonNode configuration) {
```
Comment (sqwci) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java]: MAJOR Remove this unused method parameter "configuration". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java [hunk 2]
```diff
@@ -98,9 +67,70 @@ public boolean print(byte[] content, JsonNode configuration) {
       log.error(buffer.toString());
```
Comment (sqwci) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java [hunk 3]
```diff
@@ -29,6 +30,13 @@
  * Transfer method to send data to a printer.
  */
 public enum Backend {
+  BRADY_FTP("host", "pin") {
+    @Override
+    public boolean print(byte[] content, JsonNode configuration) {
+      return sendFtpFile(configuration.get("host").asText(), "root", configuration.get("pin").asText(), "/execute",
+          content.hashCode() + ".LBL", content, configuration);
```
Comment (sqwci) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java]: MAJOR Use "Arrays.hashCode(array)" instead. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2116)

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Driver.java#L55](https://github.com/TGAC/miso-lims/blob/9166435dd55bd516db0bb1a27fe9b9aca6cd5e3f/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Driver.java#L55): Define a constant instead of duplicating this literal "A 1\n" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [Driver.java#L57](https://github.com/TGAC/miso-lims/blob/9166435dd55bd516db0bb1a27fe9b9aca6cd5e3f/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Driver.java#L57): Define a constant instead of duplicating this literal "get raw state" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Driver.java#L55](https://github.com/TGAC/miso-lims/blob/292863109af00e456b77f9cf595827c41b9b1383/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Driver.java#L55): Define a constant instead of duplicating this literal "A 1\n" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [Driver.java#L57](https://github.com/TGAC/miso-lims/blob/292863109af00e456b77f9cf595827c41b9b1383/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Driver.java#L57): Define a constant instead of duplicating this literal "get raw state" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java [hunk 4]
```diff
@@ -98,9 +68,69 @@ public boolean print(byte[] content, JsonNode configuration) {
...
+        log.error("FTP client isn't connected. Please supply a client that has connected to the host.");
+        return false;
+      }
+
+      if (directory == null) {// Do Nothing
+
+      } else if (!ftp.changeWorkingDirectory(directory)) {
+        log.error("Desired path does not exist on the server");
+        ftp.logout();
+        ftp.disconnect();
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java]: Could put logout+disconnect in a finally block connected to the try block that wraps most of this method

**Comment by sqwci** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Driver.java#L55](https://github.com/TGAC/miso-lims/blob/8c05c797b19c4bae688e291220ea88ad68dcb747/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Driver.java#L55): Define a constant instead of duplicating this literal "A 1\n" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [Driver.java#L57](https://github.com/TGAC/miso-lims/blob/8c05c797b19c4bae688e291220ea88ad68dcb747/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Driver.java#L57): Define a constant instead of duplicating this literal "get raw state" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**